src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -83,20 +83,33 @@ internal static partial class DependentTypeFinder
             // are passed in.  There is no need to check D as there's no way it could
             // contribute an intermediate type that affects A or C.  We only need to check
             // A, B and C
+            //
+            // An exception to the above rule is if we're just searching a single project.
+            // in that case there can be no intermediary projects that could add types.
+            // So we can just limit ourselves to that single project.
```

Comment:
this is an important optimization, and it ensures that if FAR asks the DependentTypeFinder to find derived symbols in a particular project that we only search in that project.